components/profile/PersonalLinksSection.js

```diff
@@ -83,9 +83,9 @@ const PersonalLinksSection = ({
   }
 
   useEffect(() => {
-    if (!profileLinks) return
-    setLinks({ type: 'overwrite', data: profileLinks })
-  }, [profileLinks])
+    if (Object.values(profileLinks ?? {}).find((p) => p.valid === false))
+      setLinks({ type: 'overwrite', data: profileLinks })
+  }, [profileLinks?.homepage])
```

Comment:
this useEffect is handling the case that profileEdit page setting an error on the profile links when the profile has no personal link at all.
and profileEdit component will only set error true to homepage (so that it can have the red border)
that's why here the dependency has only homepage.

any suggestions to make this clearer?

Reply:
OK so if I'm understanding you, the code could be just be changed to 

```js
  if (profileLinks?.homepage?.valid === false) {
    setLinks({ type: 'overwrite', data: profileLinks })
  }
```

This would make it more clear that this useEffect is only operating on the homepage field.